### DISCUSSION TITLE
very slow response times randomly after executing arbitrary commands in CLI or defocus/focus operations

### DISCUSSION THREAD

**Discussion by notjames** [ghostty-org/ghostty#4161] [Q&A]
Is anyone else experiencing issues in the following environment (at least)

* ghostty 1.0.0
* bash 5.x
* MBP 16" M3
* OSX Sonoma 14.6.1
* both in or out of tmux
* starship prompt

It's so weird! I'll run a command which runs just fine, but the prompt may take a while to return. Or the prompt returns, but if I type my next command, it takes seconds for what I type to echo to the screen. If I open interactive commands like nvim, it may take a while for me to type anything. If I take focus off ghostty to another application and then come back to ghostty, what I typed will echo to the screen, but I still won't be able to type anything.

Out of curiosity, I opened the debug screen to see if I could glean anything interesting from there (nice tool btw). What I see  baffles me. In the keyboard tab, I see that what I type is showing up in the debug tool, but nothing is happening in the terminal, so ghostty is seemingly buffering my keystrokes until it "wakes up." If I just wait for several seconds (totally random times), the ghostty comes back to life, but then if I take the focus away in those moments, that seems to also trigger ghostty's unresponsiveness. 

I'm still trying to figure out a reliable way to reproduce this, so so far I'm just articulating observations and wondering if I'm all alone in this issue.

**Comment by elerch** [comment]
You're not crazy, I'm seeing similar behavior on Debian 12 (Bookworm)/i3/zsh with single instance mode turned on. I don't use tabs, only windows since I let i3 do all that management for me. I thought I could consistently reproduce the delays (up to about 20 seconds for me) by letting a window sit idle for a while, then focus on it (with the keyboard, not the mouse). This tends to demonstrate the delay, but not consistently...sometimes things are completely normal. It happens often enough for me that I may not be able to use ghostty. But I'm also happy to help try to track this down if there's certain logs that would help.

**Reply by elerch** [reply]
Bumped across this again today. It "seemed to be working ok" for a while, and now I'm seeing the pause behavior. It seems to be reproducible if I change focus to the window using the keyboard only. I can consistently reproduce it this way:

1. Open ghostty and one other window (need not be ghostty)
2. Type in ghostty (not sure if this is necessary). Observe the characters appear
3. Use the mouse to shift focus to the other window
4. Use the **keyboard** to shift focus back to ghostty
5. Type in ghostty. The characters will not appear

Not that I am using 1.0.2-main+6ef757a, as I built ghostty myself and forgot to switch to a tagged release after cloning

Also, I had set focus-follows-mouse = true in my config. I did try setting this to false (the default), but I see the same behavior with either setting.
